sstables/mx/writer.cc

```diff
@@ -1466,7 +1466,7 @@ void writer::consume_end_of_stream() {
 
     _sst._components->statistics.contents[metadata_type::Serialization] = std::make_unique<serialization_header>(std::move(_sst_schema.header));
     seal_statistics(_sst.get_version(), _sst._components->statistics, _collector,
-        _sst._schema->get_partitioner().name(), _schema.bloom_filter_fp_chance(),
+        _sst._schema->get_partitioner().name(), _sst._schema->bloom_filter_fp_chance(),
         _sst._schema, _sst.get_first_decorated_key(), _sst.get_last_decorated_key(), _enc_stats);
```

Comment:
But it's good that you minimized the amount of changes == decreased chance of surprises when this get backported.

Reply:
The only reason I see for having a separate schema for sstables and for writers is that it allows for applying some sstable-specific schema changes faster. For example, it allows you to change bloom filter config or compression (or encryption?) config in the middle of a long streaming or compaction session, and it will be applied to newly-created sstable in that session. I have no idea how useful is that.

(But I should be the one asking you about this. It has been this way for longer than I've been a programmer).

Reply:
> The only reason I see for having a separate schema for sstables and for writers is that it allows for applying some sstable-specific schema changes faster. For example, it allows you to change bloom filter config or compression (or encryption?) config in the middle of a long streaming or compaction session, and it will be applied to newly-created sstable in that session. I have no idea how useful is that.
> 

My memory doesn't serve me well anymore. Reading streaming consumer, possibly the reason is that we want writer to use same schema of the reader, as columns might be dropped / added. So I think it's for correctness :-)